src/system.jl

```diff
@@ -387,11 +460,68 @@ Base.@kwdef mutable struct SystemDA <: System
-    increments::KeyedArray{Vector{Tuple{Float64, Float64}}, 2}
+    increments::KeyedArray{PriceSensitiveBid, 2}
     "Decrement bids time series data. `KeyedArray` where the axis keys are `bid ids x datetimes`"
-    decrements::KeyedArray{Vector{Tuple{Float64, Float64}}, 2}
+    decrements::KeyedArray{PriceSensitiveBid, 2}
     "Price sensitive load bids time series data. `KeyedArray` where the axis keys are `bid ids x datetimes`"
-    price_sensitive_loads::KeyedArray{Vector{Tuple{Float64, Float64}}, 2}
+    price_sensitive_loads::KeyedArray{PriceSensitiveBid, 2}
+end
+
+"""
+    $TYPEDEF
+
+Subtype of a `System` for modelling a purely financial style of day-ahead market clearing.
+Key differences to the physical day-ahead market clearing approach are that _all_ offers and
+bids are price sensitive and virtual participants are not distinguished from physical
+participants.
+
+Fields:
+$TYPEDFIELDS
+"""
+Base.@kwdef mutable struct FinancialSystemDA <: System
+    "`Dictionary` where the keys are bus names and the values are generator ids at that bus"
+    gens_per_bus::Dictionary{BusName, Vector{String}}
+    "`Dictionary` where the keys are bus names and the values are increment bid ids at that bus"
+    energy_offers_per_bus::Dictionary{BusName, Vector{BidName}}
+    "`Dictionary` where the keys are bus names and the values are decrement bid ids at that bus"
+    energy_bids_per_bus::Dictionary{BusName, Vector{BidName}}
+    "`Dictionary` where the keys are bus names and the values are load ids at that bus"
+    load_resources_per_bus::Dictionary{BusName, Vector{BidName}}
+
+    "Zones in the `System`, which will also include a `Requirements` entry for the market wide zone"
+    zones::Dictionary{ZoneNum, FourRequirements}
+    "Buses in the `System` indexed by bus name"
+    buses::Buses
+    "Generators in the `System` indexed by unit code"
+    generators::Generators
+    "Branches in the `System` indexed by branch name"
+    branches::Branches
+    """
+    The line outage distribution factor matrix of the system for a set of contingencies given
+    by the keys of the `Dictionary`. Each entry is a `KeyedArray` with axis keys
+    `branch names x branch on outage`
+    """
+    lodfs::Dictionary{String, KeyedArray{Float64, 2}}
+    """
+    Power transfer distribution factor of the system.  `KeyedArray` where the axis keys are
+    `branch names x bus names`
+    """
+    ptdf::Union{KeyedArray{Float64, 2}, Missing}
+
+    # Generator related time series
+    "Generator related time series data"
+    generator_time_series::GeneratorTimeSeries
+    "Generator status time series needed for the day-ahead formulation"
+    generator_status::GeneratorStatusDA
+
+    # Load time series
+    "Load time series data. `KeyedArray` where the axis keys are `load ids x datetimes`"
+    load_services::LoadServices
+
+    "Energy offers time series data. `KeyedArray` where the axis keys are `bid ids x datetimes`"
+    energy_offers::EnergyBids
```

Comment:
Should we switch the other systems to use `EnergyBids` as well? It looks like we should be able to infer the extra information required

Reply:
I think that could work.  I suppose there would be some (minimal?) overhead to storing the extra information, but the advantages of reusable code.

Reply:
Since it's repeated information and is generally read-only we could probably use an efficient data structure to avoid too much extra storage